android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java

```diff
@@ -61,11 +63,28 @@
 
   DeviceType deviceType;
 
+  Map<String, Object> constants;
+  AsyncTask<Void, Void, Map<String, Object>> futureConstants;
+
   public RNDeviceModule(ReactApplicationContext reactContext) {
+    this(reactContext, false);
+  }
+
+  public RNDeviceModule(ReactApplicationContext reactContext, boolean loadConstantsAsynchronously) {
```

Comment:
This makes it optional, but now for this method to be invoke with async load, does it need some corresponding hook in RNDeviceInfo.java? Like RNDeviceInfo needs two constructors with a similar flag, it stores the state, then when [RNDeviceInfo.createNativeModules is called](https://github.com/mcuelenaere/react-native-device-info/blob/55dc03dc465786f745ee260532891121da18547f/android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceInfo.java#L16) it sends the flag through? 

Otherwise I'm not sure how the new async style is invoked since the default (and only currently available) style from linking is to just new RNDeviceInfo and there are no other control flow entrypoints to toggle the behavior https://github.com/mcuelenaere/react-native-device-info/blob/55dc03dc46/example/android/app/src/main/java/com/example/MainApplication.java#L27

Reply:
Oh, you're right. It's because locally I'm using Turbo Modules and thus creating the RNDeviceModule myself. I'll adapt `RNDeviceInfo` to also take a constructor argument.